### PR TITLE
release notes plumbing

### PR DESCRIPTION
**Pull request by ProofOfKeags**

### CODE REVIEW

File: agent/src/Lib/Synchronizers.hs [hunk 1]
```diff
@@ -480,7 +480,7 @@ replaceDerivativeCerts :: (HasFilesystemBase sig m, Fused.Has (Error S9Error) si
 replaceDerivativeCerts = do
     sid <- getStart9AgentHostname
     let hostname = sid <> ".local"
-    tor            <- getAgentHiddenServiceUrl
+    torAddr        <- getAgentHiddenServiceUrl
```
Comment (ProofOfKeags) [inline agent/src/Lib/Synchronizers.hs:483]: This was to please an outstanding warning.

File: agent/src/Lib/Synchronizers.hs [hunk 2]
```diff
@@ -531,7 +531,7 @@ replaceDerivativeCerts = do
                           , duration          = 365
                           }
         hostname
-        tor
+        torAddr
```
Comment (ProofOfKeags) [inline agent/src/Lib/Synchronizers.hs:534]: See above.

File: agent/src/Lib/Tor.hs [hunk 3]
```diff
@@ -7,9 +7,7 @@ import           Network.HTTP.Client
 import           Network.Connection
 
 import           Lib.SystemPaths
-import           Network.HTTP.Client.TLS        ( mkManagerSettings
-                                                , newTlsManagerWith
-                                                )
+import           Network.HTTP.Client.TLS        ( mkManagerSettings )
```
Comment (ProofOfKeags) [inline agent/src/Lib/Tor.hs:10]: this was for unnecessary imports.

File: agent/src/Lib/External/Registry.hs [hunk 4]
```diff
@@ -150,12 +150,13 @@ getAppVersionForSpec appId spec = do
...
 
-getLatestAgentVersion :: (Has RegistryUrl sig m, Has (Error S9Error) sig m, MonadIO m) => m Version
+getLatestAgentVersion :: (Has RegistryUrl sig m, Has (Error S9Error) sig m, MonadIO m) => m (Version, Maybe Text)
 getLatestAgentVersion = do
     val <- registryRequest agentVersionPath
     parseOrThrow agentVersionPath val $ withObject "version response" $ \o -> do
-        v <- o .: "version"
-        pure v
+        v  <- o .: "version"
+        rn <- o .:? "release-notes"
```
Comment (ProofOfKeags) [inline agent/src/Lib/External/Registry.hs:158]: Assume `Nothing` if key not present.

**Review by afg419 — APPROVED** [review]

**Pull request merged**